test/functional/admin/export/document_controller_test.rb

```diff
@@ -146,4 +146,16 @@ class Admin::Export::DocumentControllerTest < ActionController::TestCase
     mock = MiniTest::Mock.new
     mock.expect :call, nil, [Whitehall::InternalLinkUpdater]
   end
+
+  test "calls DocumentCollectionGroupMembershipUpdater when document is marked as migrated" do
+    edition_linked_to = create(:edition_with_document, body: "Some document being migrated to Content Publisher")
+    edition_linked_to.document.update(slug: "some-document", locked: true)
+    create(:edition_with_document)
+    login_as :export_data_user
+
+    post :migrated, params: { id: edition_linked_to.document.id }, format: "json"
+
+    mock = MiniTest::Mock.new
+    mock.expect :call, nil, [Whitehall::DocumentCollectionGroupMembershipUpdater]
```

Comment:
Sorry I know this is a precedent set in a previous test that you're following.

It feels a bit strange to be checking a mock when we're in a functional test here. Should we be using this to do an end-to-end test?